src/shogun/distance/EuclideanDistance.cpp

```diff
@@ -177,3 +179,38 @@ float64_t CEuclideanDistance::distance_upper_bounded(int32_t idx_a, int32_t idx_
 
 	return result;
 }
+
+SGMatrix<float64_t> CEuclideanDistance::get_distance_matrix()
+{
+	REQUIRE(lhs->get_feature_class()==C_DENSE,
+		"Left hand side (was %s) has to be CDenseFeatures instance!\n", lhs->get_name());
+	REQUIRE(rhs->get_feature_class()==C_DENSE,
+		"Right hand side (was %s) has to be CDenseFeatures instance!\n", rhs->get_name());
+
+	REQUIRE(lhs->get_feature_type()==F_DREAL,
+		"Left hand side (was %s) has to be of double type!\n", lhs->get_name());
+	REQUIRE(rhs->get_feature_type()==F_DREAL,
+		"Right hand side (was %s) has to be double type!\n", rhs->get_name());
```

Comment:
Hey @lambday , Doesn't `get_feature_matrix` already do what we need i.e. copy only when subsets are present : https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/features/DenseFeatures.cpp#L268   ?

I have fixed other things.